src/Evolution/Systems/ScalarTensor/Tags.hpp

```diff
@@ -58,6 +61,108 @@ struct ScalarMass : db::SimpleTag {
   static constexpr bool pass_metavariables = false;
   static double create_from_options(const double mass_psi) { return mass_psi; }
 };
+
+/*!
+ * \brief Prefix tag to avoid ambiguities when observing variables with the same
+ * name in both parent systems.
+ * \note Since we also add compute tags for these quantities, we do not make
+ * this a derived class of `Tag`. Otherwise, we would have tags with repeated
+ * base tags in the `ObservationBox`.
+ */
+template <typename Tag>
+struct CSW : db::PrefixTag, db::SimpleTag {
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+
+/*!
+ * \brief Compute tag for observing the evolved variable Psi.
+ * \details This tag copies the data from the evolved variable Tag
+ * CurvedScalarWave::Tags::Psi into the wrapped tag used for observation.
+ */
+struct CSWPsiCompute : CSW<CurvedScalarWave::Tags::Psi>, db::ComputeTag {
+  using argument_tags = tmpl::list<CurvedScalarWave::Tags::Psi>;
+  using return_type = Scalar<DataVector>;
+  static constexpr void function(const gsl::not_null<return_type*> result,
+                                    const Scalar<DataVector>& psi) {
+    get(*result) = get(psi);
+  }
+  using base = CSW<CurvedScalarWave::Tags::Psi>;
+};
+
+/*!
+ * \brief Compute tag for observing the evolved variable Pi.
+ * \details This tag copies the data from the evolved variable Tag
+ * CurvedScalarWave::Tags::Pi into the wrapped tag used for observation.
+ */
+struct CSWPiCompute : CSW<CurvedScalarWave::Tags::Pi>, db::ComputeTag {
```

Comment:
I think you can combine all these into one:
```cpp
template <typename CswTag>
struct CswCompute : Csw<CswTag>, db::ComputeTag {
  using argument_tags = tmpl::list<CswTag>;
  using base = Csw<CswTag>;
  using return_type = typename base::type;
  static constexpr void function(const gsl::not_null<return_type*> result,
                                    const return_type& csw_var) {
    for (size_t i = 0; i < csw_var.size(); ++i) {
      make_const_view(make_not_null(&std::as_const((*result)[i], csw_var[i], 0, csw_var[i].size());
    }
  }
};
```

Reply:
I've never used the feature, but does this work as a reference tag?  Something like
```c++
template <typename CswTag>
struct CswReference : Csw<CswTag>, db::ReferenceTag {
  using base = Csw<CswTag>;
  using argument_tags = tmpl::list<CswTag>;
  static const typename base::type& get(const typename base::type& tensor) {
    return tensor;
  }
};
```

Reply:
I have tried both options. I only managed to get @nilsdeppe solution to work immediately. Although I left the separate wrappers for the constraints since I want to avoid also adding the compute tags from CSW directly to the ObservationBox (as this would make them available for the user who might then trigger an error if they choose to observe the bare tags and no those with the Csw wrapper.)

Not sure why the ReferenceTag didn't work in the same way as the compute tag. I was getting an error saying that CswReference<CurvedScalarWave::Tag::Psi> could not be retrieved from the observation box. Not sure why it was not trying to retrieve the base tag instead.